### ISSUE TITLE
Rule proposal: prefer `getHTML`/`setHTML` to `.innerHTML` 

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#2347]
### Description

See context for the Sanitizer API in https://fullystacked.net/innerhtml-alternatives/

### Fail

```js
main.innerHTML = '<h2>Mambo Nº2</h2>'
```

```js
console.log(main.innerHTML);
```

### Pass

```js
main.setHTML('<h2>Mambo Nº2</h2>')
```

```js
console.log(main.getHTML());
```

**Comment by DEVTomatoCake** [comment]
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/setHTML#browser_compatibility:~:text=WebView%20Android-,setHTML,-Experimental) all major browsers removed support for `<Element>.setHTML()` (though not `<Element>.getHTML()`) again, according to [Can I use](https://caniuse.com/mdn-api_element_sethtml) it's available in about 5% of browsers. It's still in the spec, however that help if all users should be able to use your website.

Why would you prefer something which breaks the site for most users, especially on recent browser releases, with no apparent benefits by itself (if you don't use sanitizers), over something which has been working since many years?

**Comment by fregante** [comment]
The linked article has more info on why. The rule might take years to be added and even then it could be left disabled by default until Chrome implements it again. Browsers are quick to update nowadays
